### DISCUSSION TITLE
GitHub Copilot could not connect to server. Extension activation failed: "read ETIMEDOUT"

### DISCUSSION THREAD

**Discussion by johnschult** [community/community#8177] [Copilot Conversations]
Running the extension on a MacBook...  Signed in fine, icon shows up, then barfs a few minutes later and says:

`GitHub Copilot could not connect to server. Extension activation failed: "read ETIMEDOUT"`

I am on a corporate proxy, but there are no restrictions for `api.github.com` or `copilot-proxy.githubusercontent.com`. I can ping and use that URL just fine. Is there some other endpoint the extension is trying to access? I have noticed this problem or similar issues with no real answers other than, reinstall the extension, which does nothing, or vague responses saying that it is problematic behind a proxy.

I can access https://copilot-proxy.githubusercontent.com/_ping and get a response in my browser. Additionally, this bit of code returns the correct ping response also:

**https-test.mjs**
```javascript
import http from 'http';
import https from 'https';

const host = 'copilot-proxy.githubusercontent.com';

http
  .request({
    // establishing a tunnel
    host: 'my.proxy.com',
    port: 8080,
    method: 'CONNECT',
    path: `${host}:443`,
  })
  .on('connect', (_res, socket, _head) => {
    https.get(
      {
        host,
        path: '/_ping',
        socket, // using a tunnel
        agent: false, // cannot use a default agent
      },
      (res) => {
        res.setEncoding('utf8');
        res.on('data', console.log);
      }
    );
  })
  .end();
```
**Command line**
```shell
❯ node https-test.mjs
{"now":1638056720,"status":"ok"}
```

**Comment by davecheney** [comment]
@johnschult it is possible that your corporate proxy is blocking access to api.github.com. What do you see in View -> Ouput; select GitHub Copilot?

**Reply by johnschult** [reply]
I can reach api.github.com fine.

**Reply by davecheney** [reply]
The error you posted earlier indicated that perhaps vscode could not contact api.GitHub.com. As I understand it vscode does not use the system proxy settings

**Reply by chnuessli** [reply]
Does anybody have a solution yet? I have set the proxy in vscode as well...

`[ERROR] [default] [2021-12-08T12:55:24.332Z] GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 140.82.121.6:443"`

**Reply by nicholasbca** [reply]
> Does anybody have a solution yet? I have set the proxy in vscode as well...
> 
> `[ERROR] [default] [2021-12-08T12:55:24.332Z] GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 140.82.121.6:443"`

same error here
`[ERROR] [default] [2021-12-17T05:06:40.247Z] GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 20.205.243.168:443"`

I have set http_proxy and https_proxy in environment variable

**Reply by johnschult** [reply]
FWIW the example code above using `api.github.com` and a path of `/` works just fine on the same computer (still on VPN) that fails to load Copilot.

**Comment by combor** [comment]
Same here:
```
ERROR] [default] [2022-01-17T10:58:55.725Z] GitHub Copilot could not connect to server. Extension activation failed: "read ETIMEDOUT"
```
but no indication which IP it tries to connect. Authentication with Github works well. I've got `https_proxy` and `https_proxy` set in my env.

**Reply by davecheney** [reply]
Sadly copilot does not support proxies at the moment. We understand this is a limitation for some users and we are working on it, however I cannot provide any timeline when we plan to address this.

**Reply by vincent-thomas** [reply]
I have the same issue! But I'm not on a proxy 🤔

**Reply by davecheney** [reply]
Can you please paste the relevant parts of View -> Output; select _GitHub Copilot_ from the pulldown.

**Reply by sayedMurtadha** [reply]
I got the same error at first and the extension worked after a few trials.
Here is what I did:
- Signed in using the same GitHub account in VSCode itself (not only while authorization)
- Enabled the auto redirection to VSCode in the authorization page.
- Reloaded VSCode window and the Copilot authorized me successfully.

I am not sure if any of these steps are required at all since they look irrelevant but there is no harm in trying :)

**Reply by PrajjwalDatir** [reply]
> Sadly copilot does not support proxies at the moment. We understand this is a limitation for some users and we are working on it, however I cannot provide any timeline when we plan to address this.

Does it support proxies now?

**Comment by lzcmian** [comment]
Copilot works on pycharm well, but does not work on vscode, prompting the following error.

`[ERROR] [default] [2022-02-05T12:24:18.109Z] GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 66.249.89.104:443"`
I guess the Firewall in China blocked this IP, but why did it work in PyCharm?

**Reply by jackhu-bme** [reply]
> Copilot works on pycharm well, but does not work on vscode, prompting the following error.
> 
> `[ERROR] [default] [2022-02-05T12:24:18.109Z] GitHub Copilot could not connect to server. Extension activation failed: "connect ETIMEDOUT 66.249.89.104:443"` I guess the Firewall in China blocked this IP, but why did it work in PyCharm?

Have you solved the problem? Me too couldn't use it in VScode. I can use it on my Macbook with a Chinese IP but not so on my school HPC server.

**Reply by so2liu** [reply]
use proxifier

**Reply by crackwitz** [reply]
proxifier? "31-day Free Trial" or "Buy Now"? no thanks. I'll wait for them to change their network communication so it doesn't assume a direct connection.

**Comment by jnvshubham7** [comment]
**how to use this solutoin please give me brief about that?**
```
copilot-proxy.githubusercontent.com
api.github.com (Also used for Self-hosted runners)
```


This error have shown in vs code when we trying to connect copilot on my collage internet which will use proxy system
`[ERROR] [default] [2022-06-16T05:43:59.030Z] GitHub Copilot could not connect to server. Extension activation failed: "getaddrinfo ENOTFOUND api.github.com"`

how to solve this problem ?
please reply me

**Reply by so2liu** [reply]
If you have a shadowsocks proxy, but copilot won't use it even with $https_proxy, you can use an app named Proxifier to proxy 20.205.243.168 with shadowsocks proxy. I fixed it five minutes ago.

**Comment by eeholmes** [comment]
For those who end up here with similar issues but not proxy server related,

* Made sure VSCode was up to date
* Right clicked my left sidebar in VSCode
* Clicked the Accounts so that accounts shows up in lower left
* Clicked on accounts and signed out of GitHub
* Removed Copilot Extension
* Closed and then re-opened VS Code
* Reinstalled copilot
* Finally it started working

Note for View > Output, the dropdown is down at the bottom (for me) in the Output tab (in the panel with Terminal tab). When copilot was not work, this showed nothing. Once it started working, then output appeared.

**Comment by guohenghai** [comment]
Ok, I fixed it in this way, I am using mac which has the boring problem is, it always lost DNS settings each time I rebooted it,
I fixed this problem by setting my DNS, problem gone
